std/algorithm/iteration.d

```diff
@@ -836,7 +838,7 @@ template each(alias pred = "a")
 
     enum isForeachBinaryIterable(R) =
         is(typeof((R r) {
-            foreach (i, ref a; r)
+            foreach (ref i, ref a; r)
```

Comment:
Yes, this is a problem. More concerning is that it mirrors the behavior of `foreach`. ```foreach(ref i, ref a ; are) i =0;``` is an infinite loop. We could do some trickery to prevent that here, but technically `each` isn't allowing anything more dangerous than `foreach`

On March 7, 2016 8:57:22 AM EST, Seb <notifications@github.com> wrote:
>> @@ -836,7 +838,7 @@ template each(alias pred = "a")
>>  
>>      enum isForeachBinaryIterable(R) =
>>          is(typeof((R r) {
>> -            foreach (i, ref a; r)
>> +            foreach (ref i, ref a; r)
>
>A quick hack might be to check whether it's a tuple
> -> if `no` then no ref
>-> if `yes` check for `r.front.index` ... (enumerate tuples are named)
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/D-Programming-Language/phobos/pull/3837/files#r55205790

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.